python/test/eager_mode/normalize_args_kwargs.py

```diff
@@ -0,0 +1,81 @@
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# Also available under a BSD-style license. See LICENSE.
+
+# RUN: %PYTHON %s | FileCheck %s
+
+
+import torch
+
+from torch_mlir.eager_mode.torch_mlir_dispatch import normalize_args_kwargs
+from torch_mlir_e2e_test.torchscript.configs import EagerModeTestConfig
+from torch_mlir_e2e_test.torchscript.framework import TestUtils, run_tests
+from torch_mlir_e2e_test.torchscript.registry import (
+    register_test_case,
+    GLOBAL_TEST_REGISTRY,
+)
+from torch_mlir_e2e_test.torchscript.reporting import report_results
+
+
+# CHECK: PASS - "should_normalize"
+@register_test_case(module_factory=lambda: None)
+def should_normalize(_, tu: TestUtils):
+    target = torch.ops.aten.max_pool2d_with_indices.default.overloadpacket
+    args = (torch.randn((1, 3, 32, 32)),)
+    kwargs = {"kernel_size": [3, 3]}
+    golden = {
+        "kernel_size": [3, 3],
+        # This is due to the schema for max_pool2d_with_indices defining
+        # the stride arg as int[2] stride=[].
+        "stride": [],
+        "padding": [0, 0],
+        "dilation": [1, 1],
+        "ceil_mode": False,
+    }
+
+    new_args, new_kwargs = normalize_args_kwargs(target, args, kwargs)
+    for arg, new_arg in zip(args, new_args):
+        assert torch.allclose(arg, new_arg)
+    for k, v in new_kwargs.items():
+        assert v == golden[k]
+
+
+# CHECK: FAIL - "shouldnt_normalize1"
+# CHECK: AssertionError: Couldn't normalize args and kwargs
+@register_test_case(module_factory=lambda: None)
+def shouldnt_normalize1(target, tu: TestUtils):
+    target = torch.ops.aten.max_pool2d_with_indices.default.overloadpacket
+    args = (tu.rand(1, 3, 32, 32),)
+    kwargs = {"kernel_size": []}
+    normalize_args_kwargs(target, args, kwargs)
+
+
+# CHECK: FAIL - "shouldnt_normalize2"
+# CHECK: AssertionError: Couldn't normalize args and kwargs
+@register_test_case(module_factory=lambda: None)
+def shouldnt_normalize2(_, tu: TestUtils):
+    target = torch.ops.aten.max_pool2d_with_indices.default.overloadpacket
+    args = (tu.rand(1, 3, 32, 32),)
+    kwargs = {"stride": []}
+    normalize_args_kwargs(target, args, kwargs)
+
+
+# CHECK: FAIL - "shouldnt_normalize3"
+# CHECK: AssertionError: Couldn't normalize args and kwargs
+@register_test_case(module_factory=lambda: None)
+def shouldnt_normalize3(_, tu: TestUtils):
+    target = torch.ops.aten.max_pool2d_with_indices.default.overloadpacket
+    args = (tu.rand(1, 3, 32, 32),)
+    kwargs = {"padding": None}
+    normalize_args_kwargs(target, args, kwargs)
+
+
+def main():
+    config = EagerModeTestConfig()
```

Comment:
why are you using the e2e framework? You should be able to just run the code you want and print out any exceptions/etc that are of interest to check.

Reply:
because the e2e framework has niceties like `register_test_case`, which will capture the name of the method and print it when reporting, as well as TestUtils which makes rng deterministic. 

both of these things can be rebuilt but isn't it better to reuse? 

as well as the fact that the adjacent FileCheck tests also [use the e2e framework](https://github.com/llvm/torch-mlir/blob/1d5b5a89e8e9ac75e4837b158fc4230eb7388b28/python/test/torchscript_e2e_test/basic.py#L17).

Reply:
Yes, those are the tests of the e2e framework *itself* :)

You can just do

# CHECK-LABEL: Description of Test
# CHECK: AssertionError: ... or whatever...
print("Description of Test")
# Do the thing